internal/controller/services/auth/resources/data-science-metrics-admin-clusterrole.tmpl.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: data-science-metrics-admin
+  labels:
+    app.kubernetes.io/managed-by: opendatahub-operator
+    platform.opendatahub.io/part-of: auth
+rules:
+  - apiGroups: ["authentication.k8s.io"]
+    resources: ["tokenreviews"]
+    verbs: ["create"]
+  - apiGroups: ["authorization.k8s.io"]
+    resources: ["subjectaccessreviews"]
+    verbs: ["create"]
+  - apiGroups: ["metrics.k8s.io"]
+    resources: ["pods", "nodes"]
+    verbs: ["get", "list"]
+  - apiGroups: [""]
+    resources: ["namespaces", "pods", "services", "services/proxy", "endpoints"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["apps"]
+    resources: ["deployments", "replicasets"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["monitoring.coreos.com"]
+    resources: ["servicemonitors", "podmonitors", "prometheusrules"]
+    verbs: ["create", "get", "list", "watch", "update", "patch", "delete"]
+  - apiGroups: ["route.openshift.io"]
+    resources: ["routes"]
+    verbs: ["get", "list", "watch"]
```

Comment:
@kahowell Jira says:
Using kube-rbac-proxy, restrict access to users having NodeMetrics access (this pattern is used in OpenShift Container Platform).
Do we need to also include rules for other resources like deployments, routes, service monitors etc.?